directord/server.py

```diff
@@ -664,11 +664,21 @@ def _query_coordination(self, job_id):
             time=600,
             job_id=job_id,
         ):
-            while len(self.return_jobs[job_id]["STDOUT"].keys()) != len(
-                self.return_jobs[job_id]["_nodes"]
-            ):
-                self.log.info("Waiting for [ %s ], QUERY to complete", job_id)
-                time.sleep(1)
+            while True:
+                stdout_nodes = len(self.return_jobs[job_id]["STDOUT"].keys())
+                total_nodes = len(self.return_jobs[job_id]["_nodes"])
+                if stdout_nodes != total_nodes:
+                    self.log.info(
+                        "Waiting for [ %s ], QUERY to complete", job_id
+                    )
+                    time.sleep(1)
+                elif self.return_jobs[job_id].get("FAILED"):
+                    self.log.critical(
+                        "Query job [ %s ] encountered failures.", job_id
+                    )
+                    return
+                else:
+                    break
```

Comment:
else break seems odd. Seems like it should be...

if stdout_nodes == total_nodes:
    break
elif self.return_jobs[job_id].get("FAILED"):
    ...
self.log.info("Waiting...")
time.sleep(0.1)